src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.MultiSegment.cs

```diff
@@ -2444,7 +2466,6 @@ private bool ConsumeMultiLineCommentMultiSegment(ReadOnlySpan<byte> localBuffer,
                                 _totalConsumed = prevTotalConsumed;
                                 return false;
                             }
-                            HasValueSequence = true;
```

Comment:
Revered this function to original state and redoing.

Reply:
@ahsonkhan I am not sure about what I am seeing. But this code change and the other similar one that we discussed in the previous comment thread [here](https://github.com/dotnet/corefx/pull/35934#discussion_r270167207) were all my attempts to solve the following sort of exception when I tried to work with `SequencePosition` based slicing.

Any change that I do from the code in this PR, I start getting such exceptions. Hence I thought of running the test scenario on `master` branch (which doesn't have my code yet) and I got similar exception there too. Following is one of the 8 failures. Can you help ascertain if this is really an issue in the master branch as well? Please advise if new issue needs to be created in GitHub for this. Also if this is indeed an issue then I believe the base algorithms need some fix before we go ahead with the changes in this PR?

```  Discovering: System.Text.Json.Tests (method display = ClassAndMethod, method display options = None)
  Discovered:  System.Text.Json.Tests (found 347 of 356 test cases)
  Starting:    System.Text.Json.Tests (parallel test collections = off, max threads = 8)
    System.Text.Json.Tests.Utf8JsonReaderTests.AllowCommentStackMismatchMultiSegment(jsonString: "{\"testA\":[1/*hi*//*bye*/, 2, 3], \"testB\": 4}", expectedWithoutComments: "testA123testB4", expectedWithComments: "testA1/*hi*//*bye*/23testB4") [FAIL]
      System.ArgumentOutOfRangeException : Specified argument was out of the range of valid values.
      Stack Trace:
        F:\vsagent\44\s\src\System.Private.CoreLib\shared\System\ReadOnlySpan.Fast.cs(286,0): at System.ReadOnlySpan`1.Slice(Int32 start, Int32 length)
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.MultiSegment.cs(2485,0): at System.Text.Json.Utf8JsonReader.ConsumeMultiLineCommentMultiSegment(ReadOnlySpan`1 localBuffer, Int32 leftOver, SequencePosition start, Int32 previousConsumed)
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.MultiSegment.cs(2280,0): at System.Text.Json.Utf8JsonReader.ConsumeCommentMultiSegment()
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.MultiSegment.cs(1563,0): at System.Text.Json.Utf8JsonReader.ConsumeNextTokenMultiSegment(Byte marker)
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.MultiSegment.cs(1534,0): at System.Text.Json.Utf8JsonReader.ConsumeNextTokenOrRollbackMultiSegment(Byte marker)
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.MultiSegment.cs(126,0): at System.Text.Json.Utf8JsonReader.ReadMultiSegment()
        D:\WinCPP\corefx\src\System.Text.Json\src\System\Text\Json\Reader\Utf8JsonReader.cs(223,0): at System.Text.Json.Utf8JsonReader.Read()
        D:\WinCPP\corefx\src\System.Text.Json\tests\Utf8JsonReaderTests.MultiSegment.cs(403,0): at System.Text.Json.Tests.Utf8JsonReaderTests.TestReadingJsonWithComments(Byte[] inputData, ReadOnlySequence`1 sequence, String expectedWithoutComments, String expectedWithComments)
        D:\WinCPP\corefx\src\System.Text.Json\tests\Utf8JsonReaderTests.MultiSegment.cs(388,0): at System.Text.Json.Tests.Utf8JsonReaderTests.AllowCommentStackMismatchMultiSegment(String jsonString, String expectedWithoutComments, String expectedWithComments)
```

Above stack trace is generated by the new lines between the comments `*** stack trace ***` in the code snippet below.

```C#
        public static void AllowCommentStackMismatchMultiSegment(string jsonString, string expectedWithoutComments, string expectedWithComments)
        {
            byte[] data = Encoding.UTF8.GetBytes(jsonString);

            var sequence = new ReadOnlySequence<byte>(data);
            TestReadingJsonWithComments(data, sequence, expectedWithoutComments, expectedWithComments);

            sequence = JsonTestHelper.GetSequence(data, 1);
            TestReadingJsonWithComments(data, sequence, expectedWithoutComments, expectedWithComments);

            // *** stack trace ***
            sequence = JsonTestHelper.GetSequence(data, 6);
            TestReadingJsonWithComments(data, sequence, expectedWithoutComments, expectedWithComments);
            // *** stack trace ***

            var firstSegment = new BufferSegment<byte>(ReadOnlyMemory<byte>.Empty);
            ReadOnlyMemory<byte> secondMem = data;
            BufferSegment<byte> secondSegment = firstSegment.Append(secondMem);
            sequence = new ReadOnlySequence<byte>(firstSegment, 0, secondSegment, secondMem.Length);
            TestReadingJsonWithComments(data, sequence, expectedWithoutComments, expectedWithComments);
        }
```

Original code: https://github.com/dotnet/corefx/blob/969270f5b9aa921b9ec956be87600d1f2e915403/src/System.Text.Json/tests/Utf8JsonReaderTests.MultiSegment.cs#L376-L391

Reply:
> Any change that I do from the code in this PR, I start getting such exceptions.

In that case, I would leave it as you have it (where there isn't a test failure).

> Following is one of the 8 failures. Can you help ascertain if this is really an issue in the master branch as well?

Can you please share the new test case which is showing the failure or a commit that I can checkout so I can repro locally myself and help investigate?

>  Please advise if new issue needs to be created in GitHub for this. Also if this is indeed an issue then I believe the base algorithms need some fix before we go ahead with the changes in this PR?

You have added more tests here. If your current PR (as is) fixes, what we believe, is a bug in master (that your new tests discovered), then that's definitely a good reason to keep this change and add the new tests. I would like to understand the cause of the failure better though.

@WinCPP, if you prefer, we can chat on gitter: https://github.com/dotnet/corefx#community

Reply:
Please find the new test case as applied on master over here https://github.com/WinCPP/corefx/tree/m1

Since our last syncup, I have been able to solve the existing issue as well, because the new code invariably needed similar fix. Please refer to latest version in PR.